nixos/modules/services/networking/tailscale.nix

```diff
@@ -0,0 +1,59 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let cfg = config.services.tailscale;
+in {
+  meta.maintainers = with maintainers; [ danderson mbaillie ];
+
+  options.services.tailscale = {
+    enable = mkEnableOption "Tailscale daemon";
+
+    package = mkOption {
+      type = types.package;
+      default = pkgs.tailscale;
+      defaultText = "pkgs.tailscale";
+      description = "The package providing Tailscale.";
+    };
+
+    port = mkOption {
+      type = types.port;
+      default = 41641;
+      description = "The port to listen on for tunnel traffic (0=autoselect).";
+    };
+  };
+
+  config = mkIf cfg.enable {
+    systemd.services.tailscale = {
+      description = "Tailscale daemon";
+      path = with pkgs; [ iproute iptables ];
+
+      after = [ "network-pre.target" ];
+      wants = [ "network-pre.target" ];
+      wantedBy = [ "multi-user.target" ];
+
+      unitConfig = {
+        StartLimitIntervalSec = 0;
+        StartLimitBurst = 0;
+      };
+
+      serviceConfig = {
+        ExecStart = ''
+          ${getBin cfg.package}/bin/tailscaled \
+          ${optionalString (cfg.port != null) "--port ${toString cfg.port}"}
+        '';
+
+        RuntimeDirectory = "tailscale";
+        RuntimeDirectoryMode = 755;
+
+        StateDirectory = "tailscale";
+        StateDirectoryMode = 750;
+
+        User = "root";
+        Group = "root";
```

Comment:
I believe these are the defaults.

Reply:
It's true that systemd will run services as user/group `root` by default, but I believe these config values are nuanced when set to root in that they also set an environment containing $USER/HOME/SHELL and a few others.

This was copied from Tailscale's systemd unit. However, perhaps surplus to requirements @danderson?

Reply:
Yeah, I think we can remove these. I set them explicitly in the upstream systemd unit because I wanted to be explicit, and also as a reminder to lock it down more and run as !root someday :).

But this was before we had an OSS bug tracker. Now Tailscale has a bug for systemd improvements (https://github.com/tailscale/tailscale/issues/77), so we can remove these. I'll remove them from upstream as well, so that these units stay in sync.

Reply:
Fixed, cheers.